### ISSUE TITLE
Shipston static basin attributes required for LSTM

### ISSUE THREAD

**Issue by shmh40** [ai4er-cdt/flood_risk_shipston#18]
As well as forcing data, it would be good to have the following **static** (unchanging over the years) basin attributes.
We would like these to be averaged over the whole Shipston basin. 
We have not yet prepared the LSTM for these, so this is lower priority atm! Plus, I suspect some of these will be very tricky to get hold of!

**Potential to be added to the model**

- [ ] Mean daily precipitation.
- [ ] aridity: Ratio of mean PET to mean precipitation.
- [ ] elev_mean: Catchment mean elevation.
- [ ] high_prec_dur: Average duration of high-precipitation events (number of consecutive days with ≥ 5 times mean daily precipitation).
- [ ] frac_snow_daily: Fraction of precipitation falling on days with temperatures below 0 ◦C.
- [ ] high_prec_freq: Frequency of high-precipitation days (≥ 5 times mean daily precipitation).
- [ ] slope_mean: Catchment mean slope.
- [ ] geol_permeability: Surface permeability (log10).
- [ ] carb_rocks_frac: Fraction of the catchment area characterized as “Carbonate sedimentary rocks”.
- [ ] clay_frac: Fraction of clay in the soil.
- [ ] Mean daily potential evapotranspiration.
- [ ] low_prec_freq: Frequency of dry days (< 1 mm d−1).
- [ ] soil_depth_pelletier: Depth to bedrock (maximum 50 m).
- [ ] p_seasonality: Seasonality and timing of precipitation. Estimated by representing annual
- [ ] forest_frac: Forest fraction.
- [ ] sand_frac: Fraction of sand in the soil.
- [ ] soil_conductivity: Saturated hydraulic conductivity.
- [ ] low_prec_dur: Average duration of dry periods (number of consecutive days with precipitation < 1 mm d−1).
- [ ] gvf_max: Maximum monthly mean of green vegetation fraction.
- [ ] gvf_diff: Difference between the maximum and minimum monthly mean of the green vegetation fraction.
- [ ] lai_diff: Difference between the max. and min. mean of the leaf area index.
- [ ] soil_porosity: Volumetric porosity.
- [ ] soil_depth_statsgo: Soil depth (maximum 1.5 m).
- [ ] lai_max: Maximum monthly mean of leaf area index.
- [ ] silt_frac: Fraction of silt in the soil.
- [ ] max_water_content: Maximum water content of the soil.
- [ ] area_gages2: Catchment area.
- [ ] precipitation and temperature as sine waves. Positive (negative) values indicate precipitation peaks during the summer (winter). Values of approx. 0 indicate uniform precipitation throughout the year.

**Comment by shmh40** [comment]
For a sense of the importance of each feature, and hence which should be prioritised, please look at the following table.
More sensitivity = more important!

![image](https://user-images.githubusercontent.com/56727418/96575574-d1f22a80-12c8-11eb-82cb-6f031f36e0e4.png)

**Comment by shmh40** [comment]
List of possible data sources (thanks Simon M/Luke/Marc)

UK water resources portal: https://eip.ceh.ac.uk/hydrology/water-resources/ [No data for the area - Ira]
Centre for Environmental Data Analysis (CEDA): https://www.ceda.ac.uk/
Environment Agency data search portal: https://data.gov.uk/search?filters%5Bpublisher%5D=Environment+Agency
National River Flow Archive: https://nrfa.ceh.ac.uk/ [Similar data to Wisky but more coarse in time - Ira]
Land Use: UK gov land use registry/Aerial Photography

Other resources:
Google Earth Engine: https://earthengine.google.com/ [looking into it - Ira, Arduin]
QGIS software: https://qgis.org/en/site/
LiDAR (?)

**Comment by Croydon-Brixton** [comment]
Helping out on gathering the required data for the shipston catchment now. 
I've formed a new branch ([here](https://github.com/ai4er-cdt/flood_risk_shipston/tree/shipston_data_collection) )where we can add data-loading scripts (potentially to our own @cam GDrives) as well as scripts to combine them in a way similar to Camels GB.

To support me, please simply commit directly on [this branch](https://github.com/ai4er-cdt/flood_risk_shipston/tree/shipston_data_collection)

**Comment by Croydon-Brixton** [comment]
### Note: 
**Time-constraints:** need the data from (ideally) 1970 to (at least) 2020 at daily resolution.
**Space-constraints:** need data for Shipston-on-stour
**Two types of data:** "Static attributes" and "forcings daily"

## Forcings (time series data)
### Which time-series attributes do we want for our model and where to find them?

- [x]  Runoff | UNKNOWN (NRFA?) | @Croydon-Brixton 
- [x]  Precipitation | EA | @Croydon-Brixton 
- [x]  Temperature | UNKNOWN | @Ira-Shokar 
- [ ]  PET(I) | UNKNOWN | @shmh40 
- [ ]  Shortwave rad | UNKNOWN (Maybe Landsat?) | @shmh40 
- [ ]  Humidity | UNKNOWN | @Ira-Shokar 
- [ ]  Long wave radiation | UNKNOWN | --> deemed not atm
- [ ]  Windspeed | UNKNOWN | @Ira-Shokar 

## Static data
### Which static attributes do we want?
[NRFA Source](https://nrfa.ceh.ac.uk/data/station/spatial/54106)
- [x]  Altitudes (Min, 10perc, 50 perc, 90 perc, max)  | NRFA | @arduinfindeis 
- [x]  Mean altitude | Google earth | @Ira-Shokar 
- [x]  Land cover (woodland, grassland, arable, urban) | NRFA | @arduinfindeis 
- [x]  Geology (permeability of the rock) | NRFA | @luke-scot 
- [ ]  Mean precipitation (60s - 90s) | NRFA | irrelevant
- [x]  Other (Dpsbar - catchment mean drainage path slope, propwet, bfihost | NRFA | @arduinfindeis 
- [ ]  Aridity | UNKNOWN | @Croydon-Brixton 
- [x]  Catchment area | NRFA | @arduinfindeis 
- [ ]  High precipitation frequency & duration | Calculate from Rainfall data | 
- [ ]  Mean PET | Calculate from PET timeseries | @shmh40

**Comment by luke-scot** [comment]
Static geological attributes, perhaps we can keep filling in this sheet with the others?
[Completed_camel_attributes.xlsx](https://github.com/ai4er-cdt/flood_risk_shipston/files/5450993/Completed_camel_attributes.xlsx)

**Comment by Croydon-Brixton** [comment]
@luke-scot Can you make it a google sheet? Then we don't need to download ;)

**Comment by luke-scot** [comment]
Google Sheet: https://drive.google.com/file/d/1pQYNbKKh-UqSUqmCoF6QbQZUAssHZyQ2/view?usp=sharing

**Comment by Croydon-Brixton** [comment]
For those gathering data from the NRFA: 

I wrote a quick interface to interact with their API more conveniently here: 
https://github.com/ai4er-cdt/flood_risk_shipston/blob/shipston_data_collection/src/data/nrfa_api.py

Simply pull the latest changes from the data_ collection branch to use it

**Comment by Croydon-Brixton** [comment]
And here's a link to the CAMELS-GB data processing: https://github.com/naddor/camels

**Comment by Croydon-Brixton** [comment]
@shmh40 Here's the code that they used to generate the mean PET and PETI and so so on
https://github.com/naddor/camels/blob/master/clim/clim_indices.R

Aridity is apparently simply "mean(pet)/ mean(percipitaiton)" (c.f. code below). 
Once you have the mean PET values let me know then we should be able to get aridity easily.
```
# aridity
p_mean<-mean(prec,na.rm=TRUE)
pet_mean<-mean(pet,na.rm=TRUE)
aridity<-pet_mean/p_mean
```

**Comment by Croydon-Brixton** [comment]
Update on the discharge_vol data:
![image](https://user-images.githubusercontent.com/19754061/97573949-67d13800-19e2-11eb-9276-b3e455b47ecd.png)

**Comment by Croydon-Brixton** [comment]
The discharge_vol problem was fixed by noting that the NRFA uses a **water day** definition of **9AM to 8.59 AM**. By shifting the timeseries for the daily average we obtain good agreement of the WISKI and NRFA data. With the new rainday definition the NRFA and Wiski data agree quite well (max error of ~0.65 m3/s on a day with mean flow of ~90 m3/s)

![image](https://user-images.githubusercontent.com/19754061/98581415-5c6eee00-22b9-11eb-9bd3-05f607d0cb13.png)

I saved the data to our datadrive as `processed_data/shipston_hydro_timeseries.csv`. Here a view of the columns:
![image](https://user-images.githubusercontent.com/19754061/98581603-a526a700-22b9-11eb-9d42-61c90a2dcb25.png)
![image](https://user-images.githubusercontent.com/19754061/98581671-bcfe2b00-22b9-11eb-816e-048e49261a4d.png)


The most reliable data column in this CSV is `discharge_vol (wiski-observed)`  which is available from January 1986 to end of June 2020.
`discharge_vol (wiski-predicted)` uses Matt's GP to extrapolate the flow values from stage measurements in the years prior to 1986 (starting with 1973) in case you also need earlier data.
